### DISCUSSION TITLE
les champs "StatusChangeDetails" et "StatusChangeReason" en double par rapport au "statusChange" ?

### DISCUSSION THREAD

**Discussion by alexisthethe** [before-interop/anomalieAdresse#12] [Q&A] [answered]
Les champs "StatusChangeDetails" et "StatusChangeReason" concernent-ils bien le DERNIER changement de status du ticket ?

Dans ce cas, ne retrouve-t-on pas aussi cette information dans la liste des changements de status dans le champ "statusChange" ? (Cela correspondrait au dernier item du tableau de "statusChange" ?)

Y a t il un intérêt à avoir cette information en double ?

**Comment by ggrebert** [comment, marked as answer]
En effet, c'est une donnée en double. C'est bien le dernier statut du ticket.
Il y a plusieurs raisons à cela:

* Héritage de TMF
  * https://datamodel.tmforum.org/en/latest/Common/TroubleTicket/
  * https://github.com/tmforum-rand/schemas/blob/candidates/Common/TroubleTicket.schema.json
* Le tableau `statusChange` est en *Read Only*.
  Pour mettre à jour un statut, c'est beaucoup plus simple pour une API Rest de modifier la valeur d'un attribut plutôt que d'ajouter un nouvel élément à un tableau. Ça évite de devoir renvoyer tous les éléments du tableau et on n'a pas besoin de connaître toute l'historique pour modifier le statut.
* C'est plus simple pour les recherches en *query param*.

**Reply by alexisthethe** [reply]
OK j'ai compris. Cela me convient. Merci pour ta réponse :)
